CesiumGS/cesium
# Fix camera collision on tileset load

# Description

Merge https://github.com/CesiumGS/cesium/pull/11829 first!

This ensures the scene is subscribed to tileset height updates, ie. when a tile loads. The subscription requires the current position, so it should only check for updates when a tile above or below the provided position loads in, and we reset the subscription when the camera position changes. 

This is a similar workflow to how clamp to ground works, which is why these functions already exist.

## Issue number and link

Fixes https://github.com/CesiumGS/cesium/issues/11824

## Testing plan

1. Go to the [Google 3D Tiles Sandcastle](https://sandcastle.cesium.com/?src=Google%20Photorealistic%203D%20Tiles.html&label=All)
2. Use the geocoder to navigate to the Grand Canyon
3. Quickly move the camera down and underground before the view is stable. The camera should no longer ever get "stuck" under terrain as it does in main.
4. Ensure consistant FPS with Google P3DT
5. Repeat the above steps with [CWT](https://sandcastle.cesium.com/?src=Cesium%20World%20Terrain.html&label=All)

# Author checklist

- [x] https://github.com/CesiumGS/cesium/pull/11829 is merged
